fdbserver/workloads/BlobRestoreWorkload.actor.cpp

```diff
@@ -106,40 +107,92 @@ struct BlobRestoreWorkload : TestWorkload {
+		state Arena arena;
 		loop {
 			try {
-				RangeResult result = wait(tr1.getRange(normalKeys, CLIENT_KNOBS->TOO_MANY));
-				data = result;
-				break;
+				Version ver = wait(tr.getReadVersion());
+				fmt::print("Read version {} \n", ver);
+				RangeResult result = wait(tr.getRange(begin, end, limits, Snapshot::True));
+				for (auto& row : result) {
+					data.push_back_deep(data.arena(), KeyValueRef(row.key, row.value));
+				}
+				if (!result.more) {
+					break;
+				}
+				if (result.readThrough.present()) {
+					begin = firstGreaterOrEqual(KeyRef(arena, result.readThrough.get()));
+				} else {
+					begin = firstGreaterThan(KeyRef(arena, result.end()[-1].key));
+				}
 			} catch (Error& e) {
-				wait(tr1.onError(e));
+				wait(tr.onError(e));
 			}
 		}
+		return data;
+	}
 
-		// Write back to original db for Cycle worker load to verify
-		state Transaction tr2(cx);
-		loop {
-			try {
-				tr2.clear(normalKeys);
-				for (auto kv : data) {
-					tr2.set(kv.key, kv.value);
+	static void compare(VectorRef<KeyValueRef> src, VectorRef<KeyValueRef> dest) {
+		if (src.size() != dest.size()) {
+			fmt::print("Size mismatch src {} dest {}\n", src.size(), dest.size());
+			int i = 0;
+			for (; i < src.size() && i < dest.size(); ++i) {
+				if (src[i].key != dest[i].key || src[i].value != dest[i].value) {
+					fmt::print("First mismatch row at {}\n", i);
+					fmt::print("  src {} = {}\n", src[i].key.printable(), src[i].value.printable());
+					fmt::print("  dest {} = {}\n", dest[i].key.printable(), dest[i].value.printable());
+					break;
 				}
-				wait(tr2.commit());
-				fmt::print("Copied {} rows to origin db\n", data.size());
-				return Void();
-			} catch (Error& e) {
-				wait(tr2.onError(e));
+			}
+
+			TraceEvent(SevError, "TestFailure")
+			    .detail("Reason", "Size Mismatch")
+			    .detail("Src", dest.size())
+			    .detail("Dest", src.size());
+
+			return;
+		}
+
+		for (int i = 0; i < src.size(); ++i) {
+			if (src[i].key != dest[i].key) {
+				fmt::print("Key mismatch at {} src {} dest {}\n", i, src[i].key.printable(), dest[i].key.printable());
+				TraceEvent(SevError, "TestFailure")
+				    .detail("Reason", "Key Mismatch")
+				    .detail("Index", i)
+				    .detail("SrcKey", src[i].key.printable())
+				    .detail("DestKey", dest[i].key.printable());
+				return;
+			}
+			if (src[i].value != dest[i].value) {
+				fmt::print("Value mismatch at {}\n", i);
+				TraceEvent(SevError, "TestFailure")
+				    .detail("Reason", "Key Mismatch")
+				    .detail("Index", i)
+				    .detail("SrcValue", src[i].value.printable())
+				    .detail("DestValue", dest[i].value.printable());
+				return;
 			}
 		}
+		fmt::print("Restore src({} rows) and dest({} rows) are consistent\n", src.size(), dest.size());
+	}
+
+	ACTOR static Future<Void> verify(Database cx, BlobRestoreWorkload* self) {
+		// restore src. data before restore
+		state Standalone<VectorRef<KeyValueRef>> src = wait(read(cx, self));
```

Comment:
Judging by this and the workload toml file, we are doing the backup and then restore after all write traffic has completed, right?
In addition to comparing the latest version, it could be potentially interesting as well to read at a previous version from the blob workers on both clusters, to ensure that restoring to a previous point in time would have the correct data. This would also capture that the backup does a proper snapshot _while_ writes are happening, instead of after
That likely isn't in scope for this PR though.

Reply:
discussed offline. it should be ok to wait backup done, then restore. we can add a new test to test against DDBalance workload